services/hanke-service/src/integrationTest/kotlin/fi/hel/haitaton/hanke/hakemus/HakemusServiceITest.kt

```diff
@@ -580,4 +610,305 @@ class HakemusServiceITest(
                 .containsExactly(newKayttaja.id)
         }
     }
+
+    @Nested
+    inner class SendHakemus {
+        private val alluId = 35124
+
+        private val areaOutsideDefaultHanke: ApplicationArea =
+            ApplicationFactory.createApplicationArea(geometry = GeometriaFactory.thirdPolygon)
+
+        @Test
+        fun `throws exception when the application doesn't exist`() {
+            val failure = assertFailure { hakemusService.sendHakemus(1234, USERNAME) }
+
+            failure.all {
+                hasClass(ApplicationNotFoundException::class)
+                messageContains("id 1234")
+            }
+        }
+
+        @Test
+        fun `throws exception when the application has been sent before`() {
+            val application =
+                hakemusFactory.builder().withMandatoryFields().withStatus(alluId = alluId).save()
+
+            val failure = assertFailure { hakemusService.sendHakemus(application.id, USERNAME) }
+
+            failure.all {
+                hasClass(ApplicationAlreadySentException::class)
+                messageContains("id=${application.id}")
+                messageContains("alluId=$alluId")
+                messageContains("status=PENDING")
+            }
+        }
+
+        @Test
+        fun `throws exception when the application fails validation`() {
+            val application =
+                hakemusFactory.builder().withMandatoryFields().withRockExcavation(null).save()
+
+            val failure = assertFailure { hakemusService.sendHakemus(application.id, USERNAME) }
+
+            failure
+                .isInstanceOf(InvalidHakemusDataException::class)
+                .messageContains("applicationData.rockExcavation")
+        }
+
+        @Test
+        fun `skips the area inside hanke check with a generated hanke`() {
```

Comment:
Maybe the name could be here also something like `skips the area-inside-hanke check...`?

Reply:
I renamed the test.